app/controllers/api/stages_controller.rb

```diff
@@ -1,11 +1,10 @@
 # frozen_string_literal: true
 class Api::StagesController < Api::BaseController
-  include CurrentProject
-
-  before_action :require_project
+  skip_before_action :require_project, only: :clone
+  before_action :production_change!, if: :production_change?, only: :update
```

Comment:
-> `ensure_non_production` and then inline the checker logic ? ... easier to read and less clutter here ...